library/core/src/intrinsics.rs

```diff
@@ -1783,27 +1783,33 @@ extern "rust-intrinsic" {
     #[rustc_nounwind]
     pub fn truncf64(x: f64) -> f64;
 
-    /// Returns the nearest integer to an `f32`. May raise an inexact floating-point exception
-    /// if the argument is not an integer.
+    /// Returns the nearest integer to an `f32`. Changing the rounding mode is not possible in Rust,
+    /// so this rounds half-way cases to the number with an even least significant digit.
+    ///
+    /// May raise an inexact floating-point exception if the argument is not an integer.
```

Comment:
This exception isn't mentioned in the round_ties_even docs -- is this something we should be guarding against somehow? Is raising an exception UB, or does that just (guaranteed) abort the program? Feels... unhelpful to have a rounding function only allow exactly integer inputs.

Reply:
This is an FP exception. All it does is set a bit in some register that Rust code is not allowed to read.

One can set up the FP environment to turn these exceptions into traps, but Rust doesn't permit that.

Reply:
Sounds good, thanks.